pkg/kubeapiserver/options/authentication.go

```diff
@@ -478,6 +495,11 @@ func (o *BuiltInAuthenticationOptions) ToAuthenticationConfig() (kubeauthenticat
 		// all known signing algs are allowed when using authentication config
 		// TODO: what we really want to express is 'any alg is fine as long it matches a public key'
 		ret.OIDCSigningAlgs = oidc.AllValidSigningAlgorithms()
+
+		if ret.AuthenticationConfig != nil && ret.AuthenticationConfig.Anonymous != nil {
+			ret.Anonymous = ret.AuthenticationConfig.Anonymous
+		}
+
 	} else if o.OIDC != nil && len(o.OIDC.IssuerURL) > 0 && len(o.OIDC.ClientID) > 0 {
 		usernamePrefix := o.OIDC.UsernamePrefix
 
```

Comment:
below, where we handle reloading of the authentication config file, ensure that the Anonymous stanza doesn't change on reload

Reply:
Done.